pandas/core/dtypes/dtypes.py

```diff
@@ -223,17 +232,24 @@ class CategoricalDtype(PandasExtensionDtype, ExtensionDtype):
     _metadata = ('categories', 'ordered')
     _cache = {}  # type: Dict[str_type, PandasExtensionDtype]
 
-    def __init__(self, categories=None, ordered=None):
+    def __init__(self, categories=None, ordered: bool = None):
         self._finalize(categories, ordered, fastpath=False)
 
     @classmethod
-    def _from_fastpath(cls, categories=None, ordered=None):
+    def _from_fastpath(cls,
+                       categories=None,
```

Comment:
Can you annotate categories?

Reply:
Only imprecisely. The most similar would be a plain ``Iterable``, but that would accept e.g. "categories='abc'", which would ideally flag as a wrong type in mypy, but that's not possible ATM.

I'm a bit apprehensive about accepting wrong types, but maybe that would be the correct compromise.

Reply:
categories should be ListLike which is ArrayLike + list + tuple + Iterable (define it in _typing)